packages/modules/notification/src/models/notification.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
+
+import {
+  BeforeCreate,
+  Entity,
+  ManyToOne,
+  OnInit,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import NotificationProvider from "./notification-provider"
+
+const NotificationProviderIdIndex = createPsqlIndexStatementHelper({
+  tableName: "notification",
+  columns: "provider_id",
+})
+
+// We don't need to support soft deletes here as this information is mainly used for auditing purposes.
+// Instead, we probably want to have a TTL for each entry, so we don't bloat the DB (and also for GDPR reasons if TTL < 30 days).
+@NotificationProviderIdIndex.MikroORMIndex()
+@Entity({ tableName: "notification" })
+// Since there is a native `Notification` type, we have to call this something else here and in a couple of other places.
+export default class NotificationModel {
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  // This can be an email, phone number, or username, depending on the channel.
+  @Property({ columnType: "text" })
+  to: string
+
+  @Property({ columnType: "text" })
+  channel: string
+
+  // The template name in the provider's system.
+  @Property({ columnType: "text" })
+  template: string
+
+  // The data that gets passed over to the provider for rendering the notification.
+  @Property({ columnType: "jsonb", nullable: true })
+  data: Record<string, unknown> | null
+
+  // This can be the event name, the workflow, or anything else that can help to identify what triggered the notification.
+  @Property({ columnType: "text", nullable: true })
+  trigger_type?: string | null
+
+  // The ID of the resource this notification is for, if applicable. Useful for displaying relevant information in the UI
+  @Property({ columnType: "text", nullable: true })
+  resource_id?: string | null
+
+  // The typeame of the resource this notification is for, if applicable, eg. "order"
+  @Property({ columnType: "text", nullable: true })
+  resource_type?: string | null
+
+  // The ID of the customer this notification is for, if applicable.
+  @Property({ columnType: "text", nullable: true })
+  customer_id?: string | null
```

Comment:
@carlos-r-l-rodrigues I am actually thinking now, should this even be a field in the module? It might be better represented as a link, don't you think?

Reply:
We discussed with Oli and decided to rename this to `receiver_id` instead as a more generic term. I also added an index for it.